glide-core/redis-rs/redis/tests/auth.rs

```diff
@@ -0,0 +1,310 @@
+            match create_connection(None, ConnectionType::Cluster, Some(&cluster_context), None)
+                .await
+                .unwrap()
+            {
+                Connection::Cluster(conn) => conn,
+                _ => panic!("Expected ClusterConnection"),
+            };
+
+        // Set the password using the unified function
+        let mut management_conn = Connection::Cluster(management_connection.clone());
+        set_password(PASSWORD, &mut management_conn).await.unwrap();
+
+        // Test that we can't connect without password
+        let connection_should_fail =
+            create_connection(None, ConnectionType::Cluster, Some(&cluster_context), None).await;
+        assert!(connection_should_fail.is_err());
+
+        // Test that we can connect with password
+        let mut connection_should_succeed = match create_connection(
+            Some(PASSWORD.to_string()),
+            ConnectionType::Cluster,
+            Some(&cluster_context),
+            None,
+        )
+        .await
+        .unwrap()
+        {
+            Connection::Cluster(conn) => conn,
+            _ => panic!("Expected ClusterConnection"),
+        };
+
+        let res: RedisResult<Value> = cmd("set")
+            .arg("foo")
+            .arg("bar")
+            .query_async(&mut connection_should_succeed)
+            .await;
+        assert_eq!(res.unwrap(), Value::Okay);
+
+        // Verify that we can retrieve the set value
+        let res: RedisResult<Value> = cmd("get")
+            .arg("foo")
+            .query_async(&mut connection_should_succeed)
+            .await;
+        assert_eq!(res.unwrap(), Value::BulkString(b"bar".to_vec()));
+
+        // Kill the connection to force reconnection
+        kill_connection(&mut management_connection).await;
+
+        // Attempt to get the value again to ensure reconnection works
+        let should_be_ok: RedisResult<Value> = cmd("get")
+            .arg("foo")
+            .query_async(&mut connection_should_succeed)
+            .await;
+        assert!(should_be_ok.is_ok());
+
+        // Replace the password in the connection
+        connection_should_succeed
+            .replace_connection_password(NEW_PASSWORD.to_string())
+            .await
+            .unwrap();
+
+        // Update the password on the server
+        let mut management_conn = Connection::Cluster(management_connection.clone());
+        set_password(NEW_PASSWORD, &mut management_conn)
+            .await
+            .unwrap();
+
+        // Test that we can't connect with the old password
+        let connection_should_fail = create_connection(
+            Some(PASSWORD.to_string()),
+            ConnectionType::Cluster,
+            Some(&cluster_context),
+            None,
+        )
+        .await;
+        assert!(connection_should_fail.is_err());
+
+        // Kill the connection to force reconnection
+        kill_connection(&mut management_connection).await;
```

Comment:
why?
dont we want to send a command now, and see if the client is able to peekup the new password and reconnect by itself? I would do:
1. set new password on the servers
2. check that GET command now fails 
3. replace the password on the client replace_connection_password
4. check that GET command now succeed

Reply:
if you set connection it doesnt kick out connected connections.